### ISSUE TITLE
[$125] Request Money - The text "Split bill with" is not translated into Spanish

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#29713]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.84.9
**Reproducible in staging?:** y 
**Reproducible in production?:** y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @ayazhussain79
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1697471385072909

## Action Performed:
Action Performed:
1. Go to the group chat.
2. Click on the "+" and then click on Split bill.
3. Enter a specific amount and complete the flow.
4. Change the language to Spanish.
5. Click on the sent split bill and then click on "Show more"
6. Check the Merchant value.

## Expected Result:
The text "Split bill with" should be translated into Spanish

## Actual Result:
The text "Split bill with" is not translated into Spanish

## Workaround:
Unknown 

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/2bfebd77-2536-489b-bfd8-56f13cb783b4

</details>

<details>
<summary>Android: mWeb Chrome</summary>

 

https://github.com/Expensify/App/assets/93399543/5091ed17-efe1-4bfd-85c1-b86395a13805

</details>

<details>
<summary>iOS: Native</summary>

 

https://github.com/Expensify/App/assets/93399543/bd43ea3b-a21a-4fd2-8e2b-942d4e5f0c8a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/640a8fe4-d69a-4bb6-ac50-a28cda5ba988

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

 

https://github.com/Expensify/App/assets/93399543/993dbcd9-f450-4a0a-8d84-6a8fc5fbecee

https://github.com/Expensify/App/assets/93399543/fd377986-0673-4e63-b19a-26b3d87f4e66

</details>

<details>
<summary>MacOS: Desktop</summary>

 

https://github.com/Expensify/App/assets/93399543/18a65f0d-cd64-4369-be68-f879338f63d9

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019e81ce43efa73a84</li>
        <li>Upwork Job ID: 1714017029487788032</li>
        <li>Last Price Increase: 2023-10-19</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @kevinksullivan</details>

**Comment by abdel-h66** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Split bill details merchant is not translated

### What is the root cause of that problem?
This is due to the split bill merchant created and saved using the user's preferred locale here 



```js
const splitTransaction = TransactionUtils.buildOptimisticTransaction(
        amount,
        currency,
        CONST.REPORT.SPLIT_REPORTID,
        comment,
        '',
        '',
        '',
        `${Localize.translateLocal('iou.splitBill')} ${Localize.translateLocal('common.with')} ${formattedParticipants} [${DateUtils.getDBTime().slice(0, 10)}]`,
        undefined,
        undefined,
        undefined,
        category,
        formattedParticipants, 
    );
```

This means that whatever language was set during the creation of that transaction will be used from now as Merchant, anywhere. 

### What changes do you think we should make in order to solve the problem?

I suggest generating the Split bill merchant dynamically on the form the front end as the following. 
In the `SplitBillDetails` component, we create the merchant string with 

`const merchant = `${props.translate('iou.splitBill')} ${props.translate('common.with')} ${arrayToString( _.map(participants, (participant) => participant.login || ''))}``

This will ensure the text is translated correctly, instead of relying on a hard-coded value. 



<img width="607" alt="dividar" src="https://github.com/abdel-h66/proposals/assets/144037189/d0c77d79-de9f-4e85-b1ac-51c2f3e03a47">


### What alternative solutions did you explore? (Optional)

N/A

**Comment by narefyev91** [comment]
Proposal from @abdel-h66 looks good to me https://github.com/Expensify/App/issues/29713#issuecomment-1765230901
🎀 👀 🎀 C+ reviewed

**Comment by abdel-h66** [comment]
The mentioned issue focuses on modifying the value before the save. This one happens when no changes are made by the user and the merchant is automatically generated.

**Comment by dangrous** [comment]
So is the merchant for a split transaction ALWAYS `bill split with [participants]`? Or can it be custom? We wouldn't want to override it [here](https://github.com/Expensify/App/blob/3bbc45e755fdc2533c1bfcccba4c77fc2a6c7891/src/pages/iou/SplitBillDetailsPage.js#L100) (which is what the proposal is, unless I'm misunderstanding) if it COULD be something else. But if it can't be, then @abdel-h66's idea works for me!

**Comment by abdel-h** [comment]
Good catch ! I just double checked. even when the user changes the merchant, that value is never taken into account. Maybe we could fix that too at the same time.

**Comment by abdel-h66** [comment]
Good catch ! I just double checked. even when the user changes the merchant, that value is never taken into account. Maybe we could fix that too at the same time.

**Comment by dangrous** [comment]
Cool. I'm going to bring this up internally for a quick discussion - your solution matches current expected behavior, which is great, but if we decide to allow custom merchants in the future, it's going to be hard to figure out what's going on. So I'm gonna see if I can get a sense of if that's ever on the roadmap or not. (To be clear - this is NOT an issue with your proposal, so don't worry! I think it just might be something people haven't thought about yet.)

**Comment by youssef-lr** [comment]
@dangrous I think we can close this, we're deprecating the "Bill split with..." merchant name, we'll be using the default one "Request" or whatever the user has entered.

**Comment by dangrous** [comment]
okay got it - will that update not have the same translation issues? If so, that's great!

**Comment by youssef-lr** [comment]
ah good point, I think the issue will persist, so I guess we need to handle the translation when creating a manual request or split and translate `Request` into Spanish if needed. Since I'm working on allowing this for split, I'll include this change in the PR for both requests and splits. Sounds good?

**Comment by youssef-lr** [comment]
Actually @dangrous, I think I'll leave this out of the scope of my PR, as we use the default merchant "Request" to also decide whether we should show the merchant or not https://github.com/Expensify/App/blob/18acd82f40153fc1e5091aea3217c532142e500e/src/components/ReportActionItem/MoneyRequestPreview.js#L171

**Comment by dangrous** [comment]
Okay cool - we'll hold on your PR and then retest/reevaluate afterwards to figure out next steps!

**Comment by dangrous** [comment]
Not overdue, on hold

**Comment by dangrous** [comment]
The backend PR for #29691 has been merged, I need to figure out whether or not there's a front end fix needed as well. Once we know that, we can get this one going again and revamp proposals based on the changes!

**Comment by dangrous** [comment]
Checking in on that held issue, we might be able to take another look at this shortly

**Comment by dangrous** [comment]
Still waiting to hear whether or not #29691 is fully complete and we can take this off of hold

**Comment by youssef-lr** [comment]
@dangrous will report back here today!

**Comment by dangrous** [comment]
Looks like we're waiting on [an App and a Web-E PR](https://github.com/Expensify/App/issues/29691#issuecomment-1787564004) to finish #29691 and then we should be ready to move forward

**Comment by dangrous** [comment]
Still on hold, looks like most of the PRs are merged so we should be able to get going shortly

**Comment by dangrous** [comment]
Still waiting on linked issue

**Comment by dangrous** [comment]
Not overdue, still on hold

**Comment by dangrous** [comment]
I'm going to make this weekly while we're waiting.

**Comment by dangrous** [comment]
Still waiting on held issue

**Comment by dangrous** [comment]
Still on hold

**Comment by dangrous** [comment]
Still on hold

**Comment by dangrous** [comment]
Okay I believe we're off hold here, as #29691 has been deployed! We should retest and update any proposals based on the new changes cc @abdel-h66 since you were working on it (long ago!) before the hold.

**Comment by DylanDylann** [comment]
I can't reproduce

**Comment by dangrous** [comment]
Yeah I'm also not seeing this. I think we can close out - was this before/after the reporting bonus change, @kevinksullivan ? Let me know what you think (and if you could double check that you're unable to reproduce too, that would be awesome!)

**Comment by dangrous** [comment]
bump when you have time to confirm we can close this - thanks @kevinksullivan!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)
